packages/email-plugin/src/plugin.ts

```diff
@@ -178,6 +211,31 @@ import {
  *
  * For all available methods of extending a handler, see the {@link EmailEventHandler} documentation.
  *
+ * ## Dynamic transport settings
+ *
+ * Instead of defining static transport settings, you can also provide a function that dynamically resolves the transport settings.
+ *
+ * @example
+ * ```ts
+ * import { defaultEmailHandlers, EmailPlugin } from '\@vendure/email-plugin';
+ *
+ * const config: VendureConfig = {
+ *   plugins: [
+ *     EmailPlugin.init({
+ *       handlers: defaultEmailHandlers,
+ *       templatePath: path.join(__dirname, 'static/email/templates'),
+ *       transport: (injector, ctx) => {
+ *         if (ctx) {
+ *           return injector.get(MyTransportService).getSettings(ctx);
+ *         } else {
+ *           return defaultSettings;
```

Comment:
People might get confused about this `defaultSettings` var that is not defined anywhere. Maybe it would be clearer to do something like:

```ts
return {
  type: 'smtp',
  host: 'smtp.example.com',
  // ... etc.
}
```

Also it add an import statement for `MyTransportService` at the top just to make clear it is coming from a local file.